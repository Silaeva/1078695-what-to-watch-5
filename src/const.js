
const filmsCount = {
  PER_STEP: 8,
  SIMILAR: 4
};

const FilmTab = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const StatusCode = {
  ADD: 1,
  REMOVE: 0
};

export {filmsCount, FilmTab, AuthorizationStatus, StatusCode};
