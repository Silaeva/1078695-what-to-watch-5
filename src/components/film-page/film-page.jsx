import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";
import FilmCardList from "../film-cards-list/film-card-list";
import LogoHeader from "../logo-header/logo-header";
import UserBlock from "../user-block/user-block";
import PageFooter from "../page-footer/page-footer";
import Tabs from "../tabs/tabs";
import {filmsCount, AppRoute} from "../../const";
import {connect} from "react-redux";

import withActiveCard from "../../hocs/with-active-card/with-active-card";
import withActiveTab from "../../hocs/with-active-tab/with-active-tab";

const FilmCardListWrapped = withActiveCard(FilmCardList);
const TabsWrapped = withActiveTab(Tabs);

const FilmPage = (props) => {
  const {films, onFilmCardClick, currentFilmId} = props;

  const currentFilm = films.find((film) => film.id === +currentFilmId);
  const {title, genre, year, image, id, bgImage, bgColor} = currentFilm;

  const similarFilms = films.filter((film) => film.genre === genre).slice(0, filmsCount.SIMILAR);

  return (

    <div>
      <section className="movie-card movie-card--full" style={{backgroundColor: bgColor}}>
        <div className="movie-card__hero">
          <div className="movie-card__bg">
            <img src={bgImage} alt={title} />
          </div>

          <h1 className="visually-hidden">WTW</h1>

          <header className="page-header movie-card__head">

            <LogoHeader />

            <UserBlock />

          </header>

          <div className="movie-card__wrap">
            <div className="movie-card__desc">
              <h2 className="movie-card__title">{title}</h2>
              <p className="movie-card__meta">
                <span className="movie-card__genre">{genre}</span>
                <span className="movie-card__year">{year}</span>
              </p>

              <div className="movie-card__buttons">
                <Link to={AppRoute.PLAYER + id} className="btn btn--play movie-card__button" type="button">
                  <svg viewBox="0 0 19 19" width="19" height="19">
                    <use xlinkHref="#play-s"></use>
                  </svg>
                  <span>Play</span>
                </Link>
                <Link to="/mylist" className="btn btn--list movie-card__button" type="button">
                  <svg viewBox="0 0 19 20" width="19" height="20">
                    <use xlinkHref="#add"></use>
                  </svg>
                  <span>My list</span>
                </Link>
                <Link to={`/films/${id}/review`} className="btn movie-card__button">Add review</Link>
              </div>
            </div>
          </div>
        </div>

        <div className="movie-card__wrap movie-card__translate-top">
          <div className="movie-card__info">
            <div className="movie-card__poster movie-card__poster--big">
              <img src={image} alt={title} width="218" height="327" />
            </div>

            <TabsWrapped film={currentFilm} />

          </div>
        </div>
      </section>

      <div className="page-content">
        <section className="catalog catalog--like-this">
          <h2 className="catalog__title">More like this</h2>

          <FilmCardListWrapped onFilmCardClick={onFilmCardClick} films={similarFilms} />

        </section>

        <PageFooter />

      </div>
    </div>
  );
};

FilmPage.propTypes = {
  currentFilmId: PropTypes.string.isRequired,
  onFilmCardClick: PropTypes.func.isRequired,
  films: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    year: PropTypes.number.isRequired,
    image: PropTypes.string.isRequired,
    bgImage: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired
  }))
};

const mapStateToProps = ({DATA}) => ({
  films: DATA.films,
});

export {FilmPage};
export default connect(mapStateToProps)(FilmPage);
