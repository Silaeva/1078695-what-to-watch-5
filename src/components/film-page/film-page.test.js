import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import {FilmPage} from "./film-page";
import {films, noop} from "../../test-data";
import {AuthorizationStatus} from "../../const";

jest.mock(`../film-card-list/film-card-list`, () => `FilmCardList`);
jest.mock(`../logo-header/logo-header`, () => `LogoHeader`);
jest.mock(`../user-block/user-block`, () => `UserBlock`);
jest.mock(`../page-footer/page-footer`, () => `PageFooter`);
jest.mock(`../tabs/tabs`, () => `Tabs`);
jest.mock(`../my-list-button/my-list-button`, () => `MyListButton`);
jest.mock(`../loading/loading`, () => `Loading`);
jest.mock(`../error-screen/error-screen`, () => `ErrorScreen`);

describe(`Should FilmPage render correctly`, () => {
  it(`When user authorised`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <FilmPage
              films={films}
              onFilmCardClick={noop}
              currentFilmId={1}
              authorizationStatus={AuthorizationStatus.AUTH}
              loadFilm={noop}
              filmById={films[0]}
              isFilmByIdLoading={false}
              isFilmByIdLoadError={false}
            />
          </BrowserRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`When user not authorised`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <FilmPage
              films={films}
              onFilmCardClick={noop}
              currentFilmId={1}
              authorizationStatus={AuthorizationStatus.NO_AUTH}
              loadFilm={noop}
              filmById={films[0]}
              isFilmByIdLoading={false}
              isFilmByIdLoadError={false}
            />
          </BrowserRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`When page loading`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <FilmPage
              films={films}
              onFilmCardClick={noop}
              currentFilmId={1}
              authorizationStatus={AuthorizationStatus.AUTH}
              loadFilm={noop}
              filmById={films[0]}
              isFilmByIdLoading={true}
              isFilmByIdLoadError={false}
            />
          </BrowserRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With page loading error`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <FilmPage
              films={films}
              onFilmCardClick={noop}
              currentFilmId={1}
              authorizationStatus={AuthorizationStatus.AUTH}
              loadFilm={noop}
              filmById={films[0]}
              isFilmByIdLoading={false}
              isFilmByIdLoadError={true}
            />
          </BrowserRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
