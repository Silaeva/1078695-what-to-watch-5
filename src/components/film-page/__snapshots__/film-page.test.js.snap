// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilmPage render correctly When page loading 1`] = `<Loading />`;

exports[`Should FilmPage render correctly When user authorised 1`] = `
<div>
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": "#795433",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Pulp Fiction"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <LogoHeader />
        <UserBlock />
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Pulp Fiction
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Crime
            </span>
            <span
              className="movie-card__year"
            >
              1994
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/1"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <MyListButton
              id={1}
              isFavorite={false}
            />
            <a
              className="btn movie-card__button"
              href="/films/1/review"
              onClick={[Function]}
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Pulp Fiction"
            height="327"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg"
            width="218"
          />
        </div>
        <Tabs
          activeTab="Overview"
          film={
            Object {
              "actors": "John Travolta, Uma Thurman, Samuel L. Jackson",
              "bgColor": "#795433",
              "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg",
              "description": "The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.",
              "director": "Quentin Tarantino",
              "duration": 153,
              "genre": "Crime",
              "id": 1,
              "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg",
              "isFavorite": false,
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg",
              "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "rating": Object {
                "count": 163242,
                "score": 1.5,
              },
              "srcVideo": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
              "title": "Pulp Fiction",
              "year": 1994,
            }
          }
          handleActiveTab={[Function]}
        />
      </div>
    </div>
  </section>
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <FilmCardList
        films={
          Array [
            Object {
              "actors": "John Travolta, Uma Thurman, Samuel L. Jackson",
              "bgColor": "#795433",
              "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg",
              "description": "The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.",
              "director": "Quentin Tarantino",
              "duration": 153,
              "genre": "Crime",
              "id": 1,
              "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg",
              "isFavorite": false,
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg",
              "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "rating": Object {
                "count": 163242,
                "score": 1.5,
              },
              "srcVideo": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
              "title": "Pulp Fiction",
              "year": 1994,
            },
            Object {
              "actors": "John Travolta, Uma Thurman, Samuel L. Jackson",
              "bgColor": "#795433",
              "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg",
              "description": "The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.",
              "director": "Quentin Tarantino",
              "duration": 153,
              "genre": "Crime",
              "id": 4,
              "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg",
              "isFavorite": false,
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg",
              "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "rating": Object {
                "count": 163242,
                "score": 1.5,
              },
              "srcVideo": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
              "title": "Pulp Fiction",
              "year": 1994,
            },
            Object {
              "actors": "John Travolta, Uma Thurman, Samuel L. Jackson",
              "bgColor": "#795433",
              "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg",
              "description": "The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.",
              "director": "Quentin Tarantino",
              "duration": 153,
              "genre": "Crime",
              "id": 7,
              "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg",
              "isFavorite": false,
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg",
              "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "rating": Object {
                "count": 163242,
                "score": 1.5,
              },
              "srcVideo": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
              "title": "Pulp Fiction",
              "year": 1994,
            },
            Object {
              "actors": "John Travolta, Uma Thurman, Samuel L. Jackson",
              "bgColor": "#795433",
              "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg",
              "description": "The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.",
              "director": "Quentin Tarantino",
              "duration": 153,
              "genre": "Crime",
              "id": 10,
              "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg",
              "isFavorite": false,
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg",
              "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "rating": Object {
                "count": 163242,
                "score": 1.5,
              },
              "srcVideo": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
              "title": "Pulp Fiction",
              "year": 1994,
            },
          ]
        }
        onFilmCardClick={[Function]}
      />
    </section>
    <PageFooter />
  </div>
</div>
`;

exports[`Should FilmPage render correctly When user not authorised 1`] = `
<div>
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": "#795433",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Pulp Fiction"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <LogoHeader />
        <UserBlock />
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Pulp Fiction
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Crime
            </span>
            <span
              className="movie-card__year"
            >
              1994
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/1"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <MyListButton
              id={1}
              isFavorite={false}
            />
            
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Pulp Fiction"
            height="327"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg"
            width="218"
          />
        </div>
        <Tabs
          activeTab="Overview"
          film={
            Object {
              "actors": "John Travolta, Uma Thurman, Samuel L. Jackson",
              "bgColor": "#795433",
              "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg",
              "description": "The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.",
              "director": "Quentin Tarantino",
              "duration": 153,
              "genre": "Crime",
              "id": 1,
              "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg",
              "isFavorite": false,
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg",
              "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "rating": Object {
                "count": 163242,
                "score": 1.5,
              },
              "srcVideo": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
              "title": "Pulp Fiction",
              "year": 1994,
            }
          }
          handleActiveTab={[Function]}
        />
      </div>
    </div>
  </section>
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <FilmCardList
        films={
          Array [
            Object {
              "actors": "John Travolta, Uma Thurman, Samuel L. Jackson",
              "bgColor": "#795433",
              "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg",
              "description": "The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.",
              "director": "Quentin Tarantino",
              "duration": 153,
              "genre": "Crime",
              "id": 1,
              "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg",
              "isFavorite": false,
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg",
              "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "rating": Object {
                "count": 163242,
                "score": 1.5,
              },
              "srcVideo": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
              "title": "Pulp Fiction",
              "year": 1994,
            },
            Object {
              "actors": "John Travolta, Uma Thurman, Samuel L. Jackson",
              "bgColor": "#795433",
              "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg",
              "description": "The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.",
              "director": "Quentin Tarantino",
              "duration": 153,
              "genre": "Crime",
              "id": 4,
              "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg",
              "isFavorite": false,
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg",
              "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "rating": Object {
                "count": 163242,
                "score": 1.5,
              },
              "srcVideo": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
              "title": "Pulp Fiction",
              "year": 1994,
            },
            Object {
              "actors": "John Travolta, Uma Thurman, Samuel L. Jackson",
              "bgColor": "#795433",
              "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg",
              "description": "The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.",
              "director": "Quentin Tarantino",
              "duration": 153,
              "genre": "Crime",
              "id": 7,
              "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg",
              "isFavorite": false,
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg",
              "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "rating": Object {
                "count": 163242,
                "score": 1.5,
              },
              "srcVideo": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
              "title": "Pulp Fiction",
              "year": 1994,
            },
            Object {
              "actors": "John Travolta, Uma Thurman, Samuel L. Jackson",
              "bgColor": "#795433",
              "bgImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg",
              "description": "The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.",
              "director": "Quentin Tarantino",
              "duration": 153,
              "genre": "Crime",
              "id": 10,
              "image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg",
              "isFavorite": false,
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg",
              "previewVideo": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "rating": Object {
                "count": 163242,
                "score": 1.5,
              },
              "srcVideo": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
              "title": "Pulp Fiction",
              "year": 1994,
            },
          ]
        }
        onFilmCardClick={[Function]}
      />
    </section>
    <PageFooter />
  </div>
</div>
`;

exports[`Should FilmPage render correctly With page loading error 1`] = `<ErrorScreen />`;
